quixstreams/checkpointing/checkpoint.py

```diff
@@ -129,7 +129,11 @@ def commit(self):
 
         # Step 2. Flush producer to trigger all delivery callbacks and ensure that
         # all messages are produced
-        self._producer.flush()
+        logger.debug("checkpoint: flushing producer")
+        failed_count = self._producer.flush()
+        if failed_count:
+            raise IncompleteFlush(failed_count)
```

Comment:
A couple of notes here:
1) The name `IncompleteFlush` doesn't clearly explain to the user that the checkpoint has timed out because of the producer. Let's rename it to `ProducerFlushTimeout` or even `CheckpointProducerTimeout` for better clarity.

2) Please add some text explaining what's going on to the exception.